### ISSUE TITLE
Supercharging session still recording all charging since december

### ISSUE THREAD

**Issue by SigLinJo** [adriankumpf/teslamate#481]
**Describe the bug**

A charging session did not get fully recorded due to some type of misscommunication between Teslamate and car. That session is still including all charging done since then when Charge Details for that charging session is opened. Se included screenshots.

**Screenshots**
What I see if I click the charging session from the 26th of december. Map removed for privacy reasons as all my charging locations since then is marked there with connecting red lines.
![image](https://user-images.githubusercontent.com/19478141/75189446-7b8b1700-574e-11ea-873a-35ac5b44bd63.png)

The offending charging session as long as it is logged. Here I manually changed the end date of displayed data.
![image](https://user-images.githubusercontent.com/19478141/75189460-83e35200-574e-11ea-8dc9-815d26ea9761.png)

Next drive starts directly after the charging without any online time inbetween:
![image](https://user-images.githubusercontent.com/19478141/75192521-a8dac380-5754-11ea-9b16-9515b3262bac.png)

The drive is from same location as the charge but with low resolution in the beginning
![image](https://user-images.githubusercontent.com/19478141/75193168-f60b6500-5755-11ea-92b7-99d6a269d1d2.png)

Maybe I was to quick to end the charge and get going for any online time to be registered and therefore no termination of the charge where made?

I'm sorry for not including more logs at this stage, but I have some trouble locating them on my manual installation. I am more than ready to try to dig up anything that can be of use, if I can get some pointers to where the Teslamate logs might be? 

My suggestion is that maybe there should be some function that terminates a charging session at the latest previously logged instance as soon as driving is logged, no matter what was recorded before...

It is a Model 3 SR+ that ran 2019.40.50.1 at the time of the offending charging session...

**Operating environment**

- TeslaMate Version: [e.g. v1.14.1 to v1.17.1]
- Type of installation: [Manual]
- OS: [FreeBSD 11.3, jail on FreeNAS]

**Comment by adriankumpf** [comment]
Since this charging session does not have an end date Grafana will preview _all_ charge data after  its start date. This has no effect on other metrics. Therefore I see no reason to "terminate" such incorrectly recorded charging sessions. You can delete the charge from the database if you want:  https://teslamate.readthedocs.io/en/latest/maintenance/manually_fixing_data.html

**Comment by adriankumpf** [comment]
Alternatively, you can manually set an end_date (assuming 999 is the id of the charging session):

```bash
docker-compose run teslamate bin/teslamate eval
"TeslaMate.Repo.get(TeslaMate.Log.ChargingProcess, 999) |> TeslaMate.Log.complete_charging_process()"
```

**Comment by SigLinJo** [comment]
Thank you for your response! I will try to fix the charging session by manually set an end date. Will the command you provided give me the opportunity to enter date and time interactively or do I need to provide it in the command somehow?

What could be done to avoid charging sessions to end up without end dates in the future? Do you have any ideas to why it happened this time?

Thank you for all the work you put into Teslamate! I appreciate it a lot and every update brings new improvements that really make it useful! The latest addition of the Trip dashboard is really nice! It was while playing with it that I came across the broken charge session and remembered that I wanted to ask you about it :) Keep up the excellent work!

**Comment by adriankumpf** [comment]
With the command (or the one from the [updated docs](https://teslamate.readthedocs.io/en/latest/maintenance/manually_fixing_data.html)) it automatically selects the end date of the last charge data associated with this charging sessionn. The same command would normally run automatically after charging.

> What could be done to avoid charging sessions to end up without end dates in the future? Do you have any ideas to why it happened this time?

Should this happen again in the future, please create a bug report with the attached logs. 
Charging sessions should always be fully recorded. If TeslaMate has not been intentionally restarted in the meantime, an incomplete charging session is most likely caused by a bug. Since the charge session you mentioned took place in December, there is a good chance that the bug that caused it to abort has already been fixed.

Without the logs for the period it is impossible to understand what happened. Therefore the logs are very important :)

>Thank you for all the work you put into Teslamate! I appreciate it a lot and every update brings new improvements that really make it useful! The latest addition of the Trip dashboard is really nice! It was while playing with it that I came across the broken charge session and remembered that I wanted to ask you about it :) Keep up the excellent work!

Thank you :)

**Comment by SigLinJo** [comment]
Hi again. As I'm learning more I found the logs for the trip where I had this strange charging session happen. Posting it if you are interested and maybe can get some understanding from it :) As stated before, when the unfinished charge happened I was running V 1.14.1 and the car ran 2019.40.50.1 at the time.

Only the missed charge:
[Trip from stockholm-charge that went wrong.txt](https://github.com/adriankumpf/teslamate/files/4424084/Trip.from.stockholm-charge.that.went.wrong.txt)

Whe whole trip that day:
[Trip from stockholm.txt](https://github.com/adriankumpf/teslamate/files/4424089/Trip.from.stockholm.txt)

And as you can se, I had a lot of connection errors back then also. I have tried to change some settings but I did not see a mesurable change this far.

Thank uou for your great work once again!

**Comment by adriankumpf** [comment]
As it happens, this was recently reported in #534 and fixed with v1.18.2 :)

**Comment by adriankumpf** [comment]
> Was the step with fixing the end date unnecessary? :)

In principle, yes, but it's no problem to do this manually beforehand  👍 

[This page](https://teslamate.readthedocs.io/en/latest/maintenance/manually_fixing_data.html) in the documentation also describes how to fix a charging session but in a slightly different way.

**Comment by SigLinJo** [comment]
Yes I found that out later in the evening and updated my post accordingly. :) The remote vorks well thou! Going through and are cleaning up charges with no duration which indicates that they where not terminated correctly. Found a bunch for the S 75 I'm managing for a friend, all sessions at his home charging location. I also had reason to use the method for combining charging sessions that you presented in #534, worked as a charm! I really like that you help us fix the data after the fact! Thanks again!
